src/backend_manager.cc

```diff
@@ -70,6 +74,16 @@ TritonBackend::Create(
 
   TritonServerMessage backend_config(backend_config_json);
 
+#ifdef TRITON_ENABLE_GPU
+  // This call is to prevent a deadlock issue with dlopening backend shared
+  // libraries and calling CUDA APIs in other threads. Since CUDA API also
+  // dlopens some libraries and dlopen has an internal lock, it can create a
+  // deadlock. Intentionally ignore the CUDA_ERROR (if any) for containers
+  // running on a CPU.
+  int device_count;
+  cudaGetDeviceCount(&device_count);
```

Comment:
Should this be moved to shared library utils to make sure all shared libraries ops have CUDA init before accessing any libraries?